Version parsing fails with master GraalVM built with latest Labs JDK 24 ea

When `native-image --version` contains this:

```
native-image 24-ea 2025-03-18
OpenJDK Runtime Environment Oracle GraalVM 24-dev.ea+10.1 (build 24-ea+10-1076)
OpenJDK 64-Bit Server VM Oracle GraalVM 24-dev.ea+10.1 (build 24-ea+10-1076, mixed mode, sharing)
```

Version parsing fails with:

```
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.11.0:build (default) on project getting-started-reactive: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.RuntimeException: Failed to get GraalVM version
[ERROR] 	at io.quarkus.deployment.pkg.steps.NativeImageBuildRunner.getGraalVMVersion(NativeImageBuildRunner.java:40)
[ERROR] 	at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:217)
[ERROR] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:588)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:1575)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:483)
[ERROR] Caused by: java.lang.NumberFormatException: For input string: "null"
[ERROR] 	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
[ERROR] 	at java.base/java.lang.Integer.parseInt(Integer.java:563)
[ERROR] 	at java.base/java.lang.Integer.parseInt(Integer.java:660)
[ERROR] 	at java.base/java.util.stream.ReferencePipeline$4$1.accept(ReferencePipeline.java:232)
[ERROR] 	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:636)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:291)
[ERROR] 	at java.base/java.util.stream.IntPipeline.toArray(IntPipeline.java:565)
[ERROR] 	at io.quarkus.deployment.pkg.steps.GraalVM$Version.breakdownVersion(GraalVM.java:231)
[ERROR] 	at io.quarkus.deployment.pkg.steps.GraalVM$Version.<init>(GraalVM.java:220)
[ERROR] 	at io.quarkus.deployment.pkg.steps.GraalVM$VersionParseHelper.parse(GraalVM.java:88)
[ERROR] 	at io.quarkus.deployment.pkg.steps.GraalVM$Version.of(GraalVM.java:297)
[ERROR] 	at io.quarkus.deployment.pkg.steps.NativeImageBuildRunner.getGraalVMVersion(NativeImageBuildRunner.java:37)
[ERROR] 	... 11 more
```

This can be replicated with this bogus test added to `GraalVMTest`:

```
        assertVersion(new Version("GraalVM 23.0.0", "23.0.0", GRAALVM), MANDREL,
            Version.of(Stream.of(("native-image 24-ea 2025-03-18\n"
                        + "OpenJDK Runtime Environment Oracle GraalVM 24-dev.ea+10.1 (build 24-ea+10-1076)\n"
                        + "OpenJDK 64-Bit Server VM Oracle GraalVM 24-dev.ea+10.1 (build 24-ea+10-1076, mixed mode, sharing)").split("\\n"))));
```

GraalVM is built with `mx build` and the latest Labs JDK which is required by oracle/graal master.

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java
@@ -15,6 +15,7 @@ public final class GraalVM {
     // Implements version parsing after https://github.com/oracle/graal/pull/6302
     static final class VersionParseHelper {
 
+        private static final String EA_BUILD_PREFIX = "-ea";
         private static final String JVMCI_BUILD_PREFIX = "jvmci-";
         private static final String MANDREL_VERS_PREFIX = "Mandrel-";
 
@@ -103,11 +104,10 @@ public final class GraalVM {
             if (vendorVersion == null) {
                 return null;
             }
-            int idx = vendorVersion.indexOf(LIBERICA_NIK_VERS_PREFIX);
-            if (idx < 0) {
+            final String version = buildVersion(vendorVersion, LIBERICA_NIK_VERS_PREFIX);
+            if (version == null) {
                 return null;
             }
-            String version = vendorVersion.substring(idx + LIBERICA_NIK_VERS_PREFIX.length());
             return matchVersion(version);
         }
 
@@ -122,11 +122,10 @@ public final class GraalVM {
             if (vendorVersion == null) {
                 return null;
             }
-            int idx = vendorVersion.indexOf(MANDREL_VERS_PREFIX);
-            if (idx < 0) {
+            final String version = buildVersion(vendorVersion, MANDREL_VERS_PREFIX);
+            if (version == null) {
                 return null;
             }
-            String version = vendorVersion.substring(idx + MANDREL_VERS_PREFIX.length());
             return matchVersion(version);
         }
 
@@ -142,17 +141,27 @@ public final class GraalVM {
             if (buildInfo == null) {
                 return null;
             }
-            int idx = buildInfo.indexOf(JVMCI_BUILD_PREFIX);
-            if (idx < 0) {
-                return null;
+            String version = buildVersion(buildInfo, JVMCI_BUILD_PREFIX);
+            if (version == null) {
+                version = buildVersion(buildInfo, EA_BUILD_PREFIX);
+                if (version == null) {
+                    return null;
+                }
             }
-            String version = buildInfo.substring(idx + JVMCI_BUILD_PREFIX.length());
             Matcher versMatcher = VERSION_PATTERN.matcher(version);
             if (versMatcher.find()) {
                 return matchVersion(version);
             } else {
                 return GRAAL_MAPPING.get(jdkFeature);
             }
+        }
+
+        private static String buildVersion(String buildInfo, String buildPrefix) {
+            int idx = buildInfo.indexOf(buildPrefix);
+            if (idx < 0) {
+                return null;
+            }
+            return buildInfo.substring(idx + buildPrefix.length());
         }
     }
 
@@ -161,8 +170,8 @@ public final class GraalVM {
     // https://github.com/quarkusio/quarkus/issues/34161
     private static final Map<Integer, String> GRAAL_MAPPING = Map.of(22, "24.0",
             23, "24.1",
-            24, "25.0",
-            25, "25.1");
+            24, "24.2",
+            25, "25.0");
 
     public static final class Version implements Comparable<Version> {
 

--- a/core/deployment/src/test/java/io/quarkus/deployment/pkg/steps/GraalVMTest.java
+++ b/core/deployment/src/test/java/io/quarkus/deployment/pkg/steps/GraalVMTest.java
@@ -166,6 +166,19 @@ public class GraalVMTest {
     }
 
     @Test
+    public void testGraalVMEA24DevVersionParser() {
+        final Version graalVMEA24Dev = Version.of(Stream.of(("native-image 24-ea 2025-03-18\n"
+                + "OpenJDK Runtime Environment Oracle GraalVM 24-dev.ea+10.1 (build 24-ea+10-1076)\n"
+                + "OpenJDK 64-Bit Server VM Oracle GraalVM 24-dev.ea+10.1 (build 24-ea+10-1076, mixed mode, sharing)")
+                .split("\\n")));
+        assertThat(graalVMEA24Dev.distribution.name()).isEqualTo("GRAALVM");
+        assertThat(graalVMEA24Dev.getVersionAsString()).isEqualTo("24.2-dev");
+        assertThat(graalVMEA24Dev.javaVersion.toString()).isEqualTo("24-ea+10-1076");
+        assertThat(graalVMEA24Dev.javaVersion.feature()).isEqualTo(24);
+        assertThat(graalVMEA24Dev.javaVersion.update()).isEqualTo(0);
+    }
+
+    @Test
     public void testGraalVMVersionsOlderThan() {
         assertOlderThan("GraalVM Version 19.3.6 CE", "GraalVM Version 20.2.0 (Java Version 11.0.9)");
         assertOlderThan("GraalVM Version 20.0.0 (Java Version 11.0.7)", "GraalVM Version 20.1.0 (Java Version 11.0.8)");

